crates/storage/provider/src/providers/static_file/writer.rs

```diff
@@ -547,6 +547,33 @@ impl StaticFileProviderRW {
         Ok(result)
     }
 
+    /// Appends multiple receipts to the static file.
+    ///
+    /// Returns the current [`TxNumber`] as seen in the static file.
+    pub fn append_receipts<I>(&mut self, receipts: I) -> ProviderResult<TxNumber>
+    where
+        I: IntoIterator<Item = (TxNumber, Receipt)>,
```

Comment:
we could avoid the vec alloc, if this accepts an iterator that returns a result instead

Reply:
accept an `IntoIterator<Item = Result<(TxNumber, Receipt), E>>`

Reply:
but IMO this is not critical because we collect into a vector only in `Receipts` static file segment, which is not in a hot path

Reply:
> accept an `IntoIterator<Item = Result<(TxNumber, Receipt), E>>`

if i am accepting an iterator returning a result, it is asking me to wrap the input into result type. most of the function are accepting an `IntoIterator` that doesn't return a result. should I still continue with this?

Reply:
yes, I think we should do it. Wrapping into a result type should be easy enough by doing `.into_iter().map(Ok)`